### PR TITLE
fix: improve amount UI entry

### PR DESCRIPTION
**Pull request by dtribble**
- preserve user text while editing
- rerender when component loses focus
- change wallet base color

### CODE REVIEW

File: packages/ui-components/src/components/NatAmountInput.jsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import { parseAsNat } from '../display/natValue/parseAsNat';
 import { stringifyNat } from '../display/natValue/stringifyNat';
-import { debounce } from '../helpers';
```
Comment (katelynsills) [inline packages/ui-components/src/components/NatAmountInput.jsx:3]: should we remove the function itself?

File: packages/ui-components/src/components/NatAmountInput.jsx [hunk 2]
```diff
@@ -47,38 +27,52 @@ const makeNatAmountInput = ({ React, TextField }) => ({
...
   };
 
-  const delayedOnChange = debounce(str => {
-    console.log('actually parsing');
-    const parsed = parseAsNat(str, decimalPlaces);
-    setDisplayString(stringifyNat(parsed, decimalPlaces, placesToShow));
-    onChange(parsed);
-  }, DEBOUNCE_WAIT_MS);
+  // Display the rendered version of the value when
+  // the user stops editign the component.
```
Comment (katelynsills) [inline packages/ui-components/src/components/NatAmountInput.jsx]:
```suggestion
  // the user stops editing the component.
```

File: packages/ui-components/src/components/NatAmountInput.jsx [hunk 3]
```diff
@@ -47,38 +27,52 @@ const makeNatAmountInput = ({ React, TextField }) => ({
...
-    // Wait until the user stops typing to parse it
-    delayedOnChange(str);
+    setFieldString(str);
+    const parsed = parseAsNat(str, decimalPlaces);
+    onChange(parsed);
   };
 
+  // If what the user is typing parses to the current
+  // value (though it might have extra punctuation),
+  // then show that rather than a computed dispaly string
```
Comment (katelynsills) [inline packages/ui-components/src/components/NatAmountInput.jsx]:
```suggestion
  // then show that rather than a computed display string
```

**Review by katelynsills — APPROVED** [review]
Nice!! 🥇

File: packages/ui-components/src/components/NatAmountInput.jsx [hunk 4]
```diff
@@ -12,15 +12,17 @@ const makeNatAmountInput = ({ React, TextField }) => ({
...
   disabled = false,
   error = false,
   onChange = () => {},
   required = false,
   helperText = null,
 }) => {
   const step = 1;
-  placesToShow = decimalPlaces > 0 ? 2 : 0;
+  if (typeof placesToShow !== 'number') {
+    placesToShow = decimalPlaces > 0 ? 2 : 0;
```
Comment (katelynsills) [inline packages/ui-components/src/components/NatAmountInput.jsx:24]: great!

File: packages/ui-components/test/components/test-NatAmountInput.js [hunk 5]
```diff
@@ -56,6 +56,8 @@ const renderNatAmountInput = ({
       error={error}
     />,
   );
+  console.log('VALUE:', result.find('input').attr('value'));
```
Comment (katelynsills) [inline packages/ui-components/test/components/test-NatAmountInput.js]: Can we remove the console log?

Comment (dtribble) [reply packages/ui-components/test/components/test-NatAmountInput.js]: done

File: packages/ui-components/src/components/NatAmountInput.jsx (hunk 1 above)
Comment (dtribble) [reply packages/ui-components/src/components/NatAmountInput.jsx:3]: We may need it still. It's a good helper to have around.

**Pull request merged**